### PR TITLE
Add front-end implementation for `conda-standalone uninstall` subcommand

### PR DESCRIPTION
**Pull request by marcoesters**
### Description

Uninstalling distributions for `conda` currently only has incomplete solutions with files and broken auto-run/initializer scripts left behind (see https://github.com/conda/constructor/issues/642, https://github.com/conda/constructor/issues/588, and https://github.com/conda/constructor/issues/572). A clean uninstallation is currently not available without manually running a series of commands that are still not 100% safe - e.g., `conda init --reverse --all` will remove initialization of `conda`, no matter what installation it is pointed to.

[conda-standalone](https://github.com/conda/conda-standalone/) is implementing an `uninstallation` subcommand (see https://github.com/conda/conda-standalone/pull/112) to alleviate these issues. This PR implements the front-end implementation of the uninstallation options for the Windows uninstaller (see screenshot).

The `conda-standalone` feature needs to be merged and uploaded first before merging this PR.

Screenshot 2024-11-12 at 11 28 26 AM

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](https://github.com/conda/constructor/blob/main/news/TEMPLATE)) for the next release's release notes?
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -164,11 +170,13 @@ def _run_uninstaller_exe(install_dir, timeout=420, check=True):
         "start",
         "/wait",
         str(uninstaller),
+        "/S",
+        *options,
```
Comment (jaimergp) [inline tests/test_examples.py:174]: I think the native NSIS flags like `/S` need to go AFTER the custom ones.

Comment (marcoesters) [reply tests/test_examples.py:174]: I think the order is "options without arguments before options with arguments".

When I ran the tests, I noticed that `/RemoveCondaRcs=user /S`, `${GetOptions} $ARGV "/RemoveCondaRcs=" $ARGV_Uninst_RemoveCondaRcs` will set `$ARGV_Uninst_RemoveCondaRcs` to `user "/S`.

Comment (jaimergp) [reply tests/test_examples.py:174]: Got it. Can we reflect these findings in the docs? I think there's a section dedicated to this aspect.

Comment (marcoesters) [reply tests/test_examples.py:174]: I updated and expanded the documentation

**Review by jaimergp — COMMENTED** [review]
The Windows changes LGTM. It's a lot of NSIS code so I'm assuming this thing runs correctly and if there are edge cases, I guess we'll hear about it.

I'm curious about the uninstallation story for Linux and macOS, though. In principle, users can just run `conda.exe constructor uninstall --prefix <install path`, right? Is it really that simple? I think we should document this in any case, maybe a new page under the How-to section? WDYT?

File: constructor/winexe.py [hunk 2]
```diff
@@ -214,6 +215,86 @@ def uninstall_menus_commands(info):
...
+                StrCpy $2 $1 "" $R0
+                StrCmp $2 ".conda_trash" advance_rmdir_loop end_rmdir
+            advance_rmdir_loop:
+                FindNext $0 $1
+                Goto rmdir_loop
+            end_rmdir_loop:
+                RMDir /r /REBOOTOK "$INSTDIR"
+            end_rmdir:
+                FindClose $0
+        skip_rmdir:
```
[the remark this answers is not in the document]
Comment (marcoesters) [reply constructor/winexe.py:288]: 🫠

**Comment by jaimergp** [comment]
> It really is that simple. A new page in the how-to section sounds interesting. We could use this to set a standard on how these uninstallers ought to be implemented. We are implicitly doing this with the Windows front-end already.

Let's write this page here and then I'll approve!

**Review by jaimergp — APPROVED** [review]

**Comment by marcoesters** [comment]
@jaimergp, I had to make a change in the uninstallation behavior. The issue was that there were some `.conda_trash` files left in a subdirectory when running on the CI and my function didn't account for it.

That gave me two choices: implement a recursive search in NSIS (no) or try and delete the directory the way we do it with the old uninstaller method. I opted for the latter.

**Review by jaimergp — APPROVED** [review]

**Pull request merged**